detox/src/devices/common/drivers/android/tools/FileXfer.js

```diff
@@ -6,6 +6,23 @@ class FileXfer {
     this._dir = destinationDir;
   }
 
+  async getFileHash(binary) {
+    const crypto = require('./CryptoUtils');
+    return await crypto.getMd5(`${this._dir}/${binary}`);
+  }
+
+  async createEmptyFile(deviceId, filename) {
```

Comment:
Why is this needed? Isn't it enough to just create a (local) hash and push it to the device?